### PR TITLE
new Boolean comparison module

### PR DESCRIPTION
**Pull request by jessdtate**
This module allows users to compare two matrices and return different outputs based on the comparison. Also a quit option.  

Some fixes to the Module ui code to enable pulling list of options from module code.

Some fixes to Module creation document.

### CODE REVIEW

File: src/Modules/Math/BooleanCompare.cc [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    std::string then_statement = state->getValue(Parameters::Then_Option).toString();
+    std::string else_statement = state->getValue(Parameters::Else_Option).toString();
+    
+    auto out_mat=output.get<DenseMatrix>(Variables::Solution);
+    double *data=out_mat->data();
+    
+    if ((data[0]==1 && then_statement == "quit") || (data[0]==0 && else_statement == "quit"))
+    {
+
+      std::cout<<"cond matrix :"<<*out_mat<<std::endl;
```
Comment (aylakhan) [inline src/Modules/Math/BooleanCompare.cc]: Log instead of print?

Comment (dcwhite) [inline src/Modules/Math/BooleanCompare.cc]: Use either `remark()` or `LOG_DEBUG`

**Comment by dcwhite** [comment]
@jessdtate Please add a test network

File: src/Modules/Math/BooleanCompare.cc [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      std::cout<<"ordered to quit.  Trying to quit."<<std::endl;
+//      quit nicely
+      std::string q_message=NetworkEditorPythonAPI::quit(false);
```
Comment (dcwhite) [inline src/Modules/Math/BooleanCompare.cc]: This part forces BUILD_WITH_PYTHON to be enabled. To fix the gcc build (which is run without python enabled), you can make your entire module require python, or just this section. 

If the former, toggle including the file in the CMakeLists and surround the whole file in `#ifdef BUILD_WITH_PYTHON`. If the latter, just surround the quit code with the same macro.

File: src/Modules/Math/BooleanCompare.cc [hunk 3]
```diff
@@ -93,12 +94,20 @@ BooleanCompare::execute()
...
 //      quit nicely
+      LOG_DEBUG("cond matrix :"<<*out_mat<<std::endl);
+      remark("ordered to quit.  Trying to quit.");
+      std::ostringstream ostr;
       std::string q_message=NetworkEditorPythonAPI::quit(false);
-      std::cout<<"Quit message = "<<q_message<<std::endl;
+      ostr << "Quit message = "<<q_message;
+      remark(ostr.str());
+#else
+      remark(" cannot quite without building with python");
```
Comment (dcwhite) [inline src/Modules/Math/BooleanCompare.cc]: s/quite/quit

**Comment by dcwhite** [comment]
I hooked up a bot that should have fixed that automatically, but I guess it's not working yet.

**Comment by dcwhite** [comment]
Still need that test network!

**Comment by dcwhite** [comment]
thanks

File: src/Interface/Modules/Math/BooleanCompareDialog.cc [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+                condition_.insert(StringPair("A is less than B (<)","lessop"));
+                condition_.insert(StringPair("A is less or equal to B (<=)","lesseqop"));
+                condition_.insert(StringPair("A is equal to B (==)","eqop"));
+                condition_.insert(StringPair("A is greater than B (>)","greatop"));
+                condition_.insert(StringPair("A is greater or equal to B (>=)","greateqop"));
+                
+                then_.insert(StringPair("Return: null","null"));
+                then_.insert(StringPair("Return: first input","first"));
+                then_.insert(StringPair("Return: second input","second"));
+                then_.insert(StringPair("Return: thrid input","third"));
```
Comment (dcwhite) [inline src/Interface/Modules/Math/BooleanCompareDialog.cc]: `s/thrid/third/`

**Pull request merged**